### PR TITLE
Link pull image secret to service accounts

### LINKED ISSUE #4900: Update Operator to link service accounts if a given secret 'syndesis-pull-secret' exists
## Minimal required

The operator should, after he installed the templates at https://github.com/syndesisio/syndesis/blob/fc7ad4c66d819759e24b3f293de9887e981eb81f/install/operator/pkg/syndesis/action/install.go#L89 link the service accounts if a given secret in the operator namespace exist.

**The secret will be called "syndesis-pull-secret".** 

## Optional enhancements 

One could go one step further to add a boolean field to the CRD like 'enablePullAuthentication' which could be used to make this behaviour more explicit. I.e. if this flag is set but the secret is missing, that an error is thrown. If this flag is not set and the secret exists, the secret is ignored.

I suggest implementing this behaviour for 7.4+.

Also one could make this kind of SA update more dynamic and e.g. monitor the CR all the time and when this flag changes in the CR then SA gets adapted. This should be implemented when the CR update functionality is implemented via #3422 .

An advantage of this post-adaption approach (create templates with SAs, modifying SAs afterwards) would be, that the template would not have to be adapted and can be directly used again on OpenShift Online without further change.

// cc @heiko-braun @avano

### LINKED ISSUE #4902: Allow for authenticated S2I builds of the integration runtimes
If pull authentication is enabled, the S2I build for the runtimes need to be able to pull the s2i builder image with authentication.

This is supposed to work with these alternative solutions:

* Add a SA to the "Build" or "BuildConfig" object, which is linked to the secret used for authentication. However it seems that this doesn't work right now.
* Add pull secret to the BuildConfig created in https://github.com/syndesisio/syndesis/blob/ba937b8043a18a90019b5b354f19ff3b10ae033b/app/server/openshift/src/main/java/io/syndesis/server/openshift/OpenShiftServiceImpl.java#L314-L325

The later approach seems to work (as @avano tested this), but it would require massive changes in the operator and installation scripts. We should strive to get the first solution running, as this allows to switch on authentication on the outside. The SA configured there should be also empowered by install_ocp.sh as described in #4900 .

### PR DESCRIPTION
**Pull request by astefanutti**
Fixes #4900
Fixes #4902

### CODE REVIEW

**Comment by heiko-braun** [comment]
@lgarciaaco Mind a review?

**Comment by heiko-braun** [comment]
@astefanutti Did you do a PR for master as well?

File: install/operator/pkg/syndesis/action/install.go [hunk 1]
```diff
@@ -91,6 +91,12 @@ func (a *installAction) Execute(ctx context.Context, syndesis *v1alpha1.Syndesis
 		})
 	}
 
+	secret := &corev1.Secret{}
+	err = a.client.Get(ctx, types.NamespacedName{Namespace: syndesis.Namespace, Name: SyndesisPullSecret}, secret)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
```
[the remark this answers is not in the document]
Comment (astefanutti) [reply install/operator/pkg/syndesis/action/install.go]: Yes, your understanding is correct. If the secret is not present, we continue the install and skip the image pull secret linking. Any other error is unexpected and thus abort the reconcile loop.

**Comment by astefanutti** [comment]
@heiko-braun not yet. I was waiting for #4930 to be ported to `master` first.

For now the list of PR to be ported to master are:
* #4930
* #4950
* #4951 (this one)

Do you think we should wait for them to be tested or we can go for porting them to `master`?

**Comment by heiko-braun** [comment]
I thikn we should have them on master first, test them and then backport.

**Comment by astefanutti** [comment]
Agreed, let me create the PRs on master.

**Comment by astefanutti** [comment]
Closing in favour of #4963.

**Pull request closed without merging**